node/core/backing/src/lib.rs

```diff
@@ -0,0 +1,1149 @@
+use std::convert::TryFrom;
+
+use bitvec::vec::BitVec;
+use log;
+use futures::{
+	select, FutureExt, SinkExt, StreamExt,
+	channel::{oneshot, mpsc},
+	future::{self, Either},
+	task::{Spawn, SpawnExt},
+};
+use futures_timer::Delay;
+use streamunordered::{StreamUnordered, StreamYield};
+
+use primitives::Pair;
+use keystore::KeyStorePtr;
+use polkadot_primitives::{
+	Hash, BlockNumber,
+	parachain::{
+		AbridgedCandidateReceipt, BackedCandidate, Id as ParaId, ValidatorPair, ValidatorId,
+		ValidatorIndex, HeadData, ValidationCode, SigningContext, PoVBlock,
+	},
+};
+use polkadot_node_primitives::{Statement, SignedFullStatement, MisbehaviorReport};
+use polkadot_subsystem::{
+	FromOverseer, OverseerSignal, Subsystem, SubsystemContext, SpawnedSubsystem,
+};
+use polkadot_subsystem::messages::{
+	AllMessages, AvailabilityStoreMessage, CandidateBackingMessage, SchedulerRoster,
+	RuntimeApiMessage, RuntimeApiRequest, CandidateValidationMessage, ValidationFailed,
+	StatementDistributionMessage, NewBackedCandidate, ProvisionerMessage, ProvisionableData,
+};
+use statement_table::{
+	generic::AttestedCandidate as TableAttestedCandidate,
+	Table, Context as TableContextTrait, Statement as TableStatement,
+	SignedStatement as TableSignedStatement,
+};
+
+#[derive(Debug, derive_more::From)]
+enum Error {
+	#[from]
+	ValidationFailed(ValidationFailed),
+	#[from]
+	Oneshot(oneshot::Canceled),
+	#[from]
+	Mpsc(mpsc::SendError),
+}
+
+/// Holds all data needed for candidate backing job operation.
+struct CandidateBackingJob {
+	/// The keystore which holds the signing keys.
+	keystore: KeyStorePtr,
+	/// The hash of the relay parent on top of which this job is doing it's work.
+	parent: Hash,
+	/// Inbound message channel receiving part.
+	rx_to: mpsc::Receiver<ToJob>,
+	/// Inbound message sending part. Handed out to whom it may concern.
+	tx_to: mpsc::Sender<ToJob>,
+	/// Outbound message channel sending part.
+	tx_from: mpsc::Sender<FromJob>,
+
+	/// The validation codes of the `ParaId`s we are assigned as validators to.
+	validation_code: ValidationCode,
+	/// `HeadData`s of the parachains that this validator is assigned to.
+	head_data: HeadData,
+	/// The `ParaId`s assigned to this validator.
+	assignment: ParaId,
+	/// We issued `Valid` statements on about these candidates.
+	issued_validity: HashSet<Hash>,
+	/// We sent `NewBackedCandidate` on these candidates.
+	backed_candidates: HashSet<Hash>,
+	/// `Some(h)` if this job has already issues `Seconded` statemt for some candidate with `h` hash.
+	seconded: Option<Hash>,
+	/// Registered backing watchers.
+	backing_watchers: Vec<mpsc::Sender<NewBackedCandidate>>,
+	/// We have already reported misbehaviors for these validators.
+	reported_misbehavior_for: HashSet<ValidatorIndex>,
+
+	table: Table<TableContext>,
+	table_context: TableContext,
+}
+
+const fn needed_votes(n_validators: usize) -> usize {
+	let mut threshold = (n_validators * 2) / 3;
+	threshold += (n_validators * 2) % 3;
```

Comment:
I believe this should be simple majority as opposed to 2/3 majority within the group.

Reply:
The name `needed_votes` could be more descriptive to that end